pxr/usd/bin/usdresolve/usdresolve.py

```diff
@@ -0,0 +1,97 @@
+#!/pxrpythonsubst
+#
+# Copyright 2016 Pixar
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+import argparse, sys, os
+
+def _Msg(msg):
+    sys.stdout.write(msg + '\n')
+
+def _Err(msg):
+    sys.stderr.write(msg + '\n')
+
+def _ConfigureAssetResolver(args, resolver):
+    if args.configureAssetPath:
+        resolver.ConfigureResolverForAsset(args.configureAssetPath)
+        resolverContext = resolver.CreateDefaultContextForAsset(args.configureAssetPath)
+    else:
+        resolver.ConfigureResolverForAsset(args.inputPath)
+        resolverContext = resolver.CreateDefaultContextForAsset(args.inputPath)
+    return resolverContext
+
+def _AnchorRelativePath(args, resolver):
```

Comment:
- ArResolver::Resolve is expected to only deal with files and not directories(*), since not all asset systems may necessarily have a notion of a directory. So I think we shouldn't call Resolve on the anchor path or os.getcwd() here. Instead, I think we should just call AnchorRelativePath on the paths as given by the user, and let the call to Resolve later on deal with it.

- I wonder if it would be better to only anchor the path if the user explicitly specifies args.anchorPath and not force the use of os.getcwd(). That way, users can see what Resolve returns for a given inputPath without the script adding on extra behavior. If they want to anchor against cwd, they could do something like ```usdresolve --anchorPath `pwd` ```.

So I think we'd be left with:
```
def _AnchorRelativePath(args, resolver):
    if args.anchorPath and resolver.IsRelativePath(args.inputPath):
       return resolver.AnchorRelativePath(args.anchorPath, args.inputPath)
    return args.inputPath
```